File: backend/src/main/java/ma/zs/alc/service/facade/admin/recomrecla/RecommendTeacherAdminService.java
package ma.zs.alc.service.facade.admin.recomrecla;

import java.util.List;
import ma.zs.alc.bean.core.recomrecla.RecommendTeacher;
import ma.zs.alc.dao.criteria.core.recomrecla.RecommendTeacherCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface RecommendTeacherAdminService extends  IService<RecommendTeacher,RecommendTeacherCriteria>  {

    List<RecommendTeacher> findByProfId(Long id);
    int deleteByProfId(Long id);
    long countByProfRef(String ref);




}

File: frontend-bqckup/src/app/shared/service/admin/recomrecla/RecommendTeacherAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {RecommendTeacherDto} from '../../../model/recomrecla/RecommendTeacher.model';
import {RecommendTeacherCriteria} from '../../../criteria/recomrecla/RecommendTeacherCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class RecommendTeacherAdminService extends AbstractService<RecommendTeacherDto, RecommendTeacherCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/recommendTeacher/';
    }

    public constrcutDto(): RecommendTeacherDto {
        return new RecommendTeacherDto();
    }

    public constrcutCriteria(): RecommendTeacherCriteria {
        return new RecommendTeacherCriteria();
    }
}
